src/qt/sigmacoincontroldialog.cpp

```diff
@@ -571,30 +579,44 @@ void SigmaCoinControlDialog::updateLabels(WalletModel *model, QDialog* dialog)
             if (fAllowFree && nBytes <= MAX_FREE_TRANSACTION_CREATE_SIZE)
                 nPayFee = 0;
 
-        if (nPayAmount > 0)
-        {
+        if (nPayAmount > 0) {
             nChange = nAmount - nPayAmount;
             if (!SigmaCoinControlDialog::fSubtractFeeFromAmount)
                 nChange -= nPayFee;
 
-            // Never create dust outputs; if we would, just add the dust to the fee.
-            if (nChange > 0 && nChange < MIN_CHANGE)
-            {
-                CTxOut txout(nChange, (CScript)std::vector<unsigned char>(24, 0));
-                if (txout.IsDust(::minRelayTxFee))
-                {
-                    if (SigmaCoinControlDialog::fSubtractFeeFromAmount) // dust-change will be raised until no dust
-                        nChange = txout.GetDustThreshold(::minRelayTxFee);
-                    else
-                    {
-                        nPayFee += nChange;
-                        nChange = 0;
+            if (fMintTabSelected) {
+                // Never create dust outputs; if we would, just add the dust to the fee.
+                if (nChange > 0 && nChange < MIN_CHANGE) {
+                    CTxOut txout(nChange, (CScript) std::vector<unsigned char>(24, 0));
+                    if (txout.IsDust(::minRelayTxFee)) {
+                        if (SigmaCoinControlDialog::fSubtractFeeFromAmount) // dust-change will be raised until no dust
+                            nChange = txout.GetDustThreshold(::minRelayTxFee);
+                        else {
+                            nPayFee += nChange;
+                            nChange = 0;
+                        }
+                    }
+                }
+
+                if (nChange == 0 && !SigmaCoinControlDialog::fSubtractFeeFromAmount)
+                    nBytes -= 34;
+
+            } else { //adding sizes of automatic remints
+                CAmount remChange = nChange;
+                std::vector <int64_t> allDenoms;
+                sigma::GetAllDenoms(allDenoms);
+                for (int64_t denom : allDenoms) {
+                    while (remChange >= denom) {
+                        nBytes += 44; //is the siz of each vout containing sigma mint
+                        remChange -= denom;
```

Comment:
you can use the function SelectMintCoinsForAmount in `wallet/wallet.h` instead.

Reply:
good catch, resolved.